mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/EntityInjectorGenerator.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.dao;
+
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.api.mapper.annotations.EntityInjector;
+import com.datastax.oss.driver.internal.mapper.processor.PartialClassGenerator;
+import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.datastax.oss.driver.internal.mapper.processor.SkipGenerationException;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeSpec;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+public class EntityInjectorGenerator implements PartialClassGenerator {
+
+  private final ExecutableElement methodElement;
+  private final DaoImplementationGenerator daoImplementationGenerator;
+  private final String entityParameterName;
+  private final TypeElement entityElement;
+  private final String targetParameterName;
+
+  public EntityInjectorGenerator(
+      ExecutableElement methodElement,
+      DaoImplementationGenerator daoImplementationGenerator,
+      ProcessorContext context) {
+    this.methodElement = methodElement;
+    this.daoImplementationGenerator = daoImplementationGenerator;
+    // We're expecting a void method, where one parameter is an annotated entity, and the other a
+    // SettableByName or subtype
+    TypeMirror returnType = methodElement.getReturnType();
+    if (returnType.getKind() != TypeKind.VOID || methodElement.getParameters().size() != 2) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "%s methods must have two parameters and a void return type",
+              EntityInjector.class.getSimpleName());
+      throw new SkipGenerationException();
```

Comment:
Don't we want to include a message in the `SkipGenerationException`?

Reply:
The message is passed to `context.getMessager()` on the previous line. It will materialize as a compiler error. It's a bit more convenient to do it here because we have all the context needed, in particular the `Element`, which determines the line of the error.

`SkipGenerationException` is just a simple way to abort quickly when we know that the generation won't succeed. The convention is that the code that throws it should take care of any message, the catching code will just ignore it and proceed (in that case it's `DaoImplementationGenerator.java:97`). It's a bit hacky, but convenient.

Note that we can never throw exceptions from processor code, errors have to be reported via the messager.